CAMEL-15822: [camel-file] fileExist=Move doesn't use FileMoveExisting…

luigidemasi opened the pull request:
…Strategy when tempFile is also configured

- [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/master/CONTRIBUTING.md

omarsmak commented on components/camel-file/src/main/java/org/apache/camel/component/file/GenericFileDefaultMoveExistingFileStrategy.java:
```diff
@@ -49,14 +49,24 @@ public boolean moveExistingFile(GenericFileEndpoint endpoint, GenericFileOperati
                     "moveExisting evaluated as empty String, cannot move existing file: " + fileName);
         }
 
+        if (to.length() > 1 && to.endsWith("/")) {
```
I see some of this logic to check the name being repeated across the three strategies, can we somehow unify this logic into a common function to ease maintainability?

luigidemasi replied:
the problem is where to put the common logic, FileUtils doesn't seem a good fit.

omarsmak replied:
By creating a singleton utility class in the component perhaps?

davsclaus reviewed (approved):
I am okay with the little duplicate code. However it would be good with a bit of code comments (eg // xxxx) what the purpose is for the code, as it does a bit of path manipulation but it may not be so easy to know why (and git blame may get harder to track down over time)

luigidemasi commented:
> I am okay with the little duplicate code. However it would be good with a bit of code comments (eg // xxxx) what the purpose is for the code, as it does a bit of path manipulation but it may not be so easy to know why (and git blame may get harder to track down over time)

I did both, I moved the duplicated code in a common utility class and added the javadoc, hope it make sense.

davsclaus commented:
Yeah this is good
